simple/cm_data.c

```diff
@@ -288,6 +301,19 @@ static int client_expect_accept(size_t paramlen)
 		return ret;
 
 	fi_shutdown(ep, 0);
+	memset(entry, 0, sizeof(*entry) + paramlen);
+	ret = fi_eq_sread(eq, &event, entry, sizeof(*entry), -1, 0);
+	if (ret < 0) {
+		fprintf(stderr, "fi_eq_sread %d\n", ret);
+		FT_PROCESS_EQ_ERR(ret, eq, "fi_eq_sread", "shutdown");
+		return ret;
+	}
+	if (event != FI_SHUTDOWN || entry->fid != &ep->fid) {
+		FT_ERR("Unexpected CM event %d fid %p (ep %p)", event,
+			entry->fid, ep);
+		ret = -FI_EOTHER;
+		return ret;
+	}
```

Comment:
Can we move these two blocks of code into a function, rather than repeating them?